src/ray/core_worker/task_manager.cc

```diff
@@ -118,6 +135,34 @@ void TaskManager::CompletePendingTask(const TaskID &task_id,
   ShutdownIfNeeded();
 }
 
+void TaskManager::CompletePendingTask(const TaskID &task_id,
+                                      const rpc::PushTaskReply &reply,
+                                      const rpc::Address &worker_addr) {
+  // Make sure that task_id doesn't correspond to an actor task.
```

Comment:
This seems like a very heavyweight check - is it necessary/is there some way to do it without grabbing & releasing the lock an additional time?